rest-api/dao/physical_measurements_dao.py

```diff
@@ -553,3 +597,75 @@ def from_client_json(resource_json, participant_id=None, **unused_kwargs):
                                 createdUsername=created_username, finalizedSiteId=finalized_site_id,
                                 finalizedUsername=finalized_username)
 
+  @staticmethod
+  def _validate_patch_update(measurement, resource):
+    """validates request of resource"""
+    cancelled_required_fields = ['status', 'reason', 'cancelledInfo']
+    restored_required_fields = ['status', 'reason', 'restoredInfo']
+
+    if resource['status'].lower() == 'cancelled':
+      if measurement.status == PhysicalMeasurementsStatus.CANCELLED:
+        raise BadRequest('This order is already cancelled')
+      for field in cancelled_required_fields:
+        if field not in resource:
+          raise BadRequest('%s is required in cancel request.' % field)
+
+    elif resource['status'].lower() == 'restored':
+      if measurement.status != PhysicalMeasurementsStatus.CANCELLED:
+        raise BadRequest('Can not restore an order that is not cancelled.')
+      for field in restored_required_fields:
+        if field not in resource:
+          raise BadRequest('%s is required in restore request.' % field)
+    else:
+      raise BadRequest('status is required in restore request.')
+
+  def _validate_and_get_author_and_site(self, resource):
+    """returns author and site based on resource cancelledInfo/restoredInfo. Validation that
+    these exists is handled by _validate_patch_update"""
+    site_id = None
+    author = None
+
+    if 'cancelledInfo' in resource:
+      site_id = self.get_location_site_id(_LOCATION_PREFIX + resource['cancelledInfo']['site'][
+        'value'])
+      author = self.get_author_username(_AUTHOR_PREFIX + resource['cancelledInfo']['author'][
+        'value'])
+
+    elif 'restoredInfo' in resource:
+      site_id = self.get_location_site_id(_LOCATION_PREFIX + resource['restoredInfo']['site'][
+        'value'])
+      author = self.get_author_username(_AUTHOR_PREFIX + resource['restoredInfo']['author'][
+        'value'])
+
+    return site_id, author
+
+  def patch(self, id_, p_id):
+    # pylint: disable=unused-argument
+    resource = request.get_json(force=True)
```

Comment:
This is a DAO. We don't normally access request in DAOs, because we want to be able to use them outside of the context of an inbound request. We should follow the pattern of other DAOs, and pass in the resource, e.g.

https://github.com/all-of-us/raw-data-repository/blob/f05196a89dbcfe631c601929ff632430c303659a/rest-api/dao/biobank_order_dao.py#L437